lib/ui5Framework/AbstractResolver.js

```diff
@@ -258,9 +242,60 @@ class AbstractResolver {
 				`Could not resolve framework version ${version}. ` +
 				`Make sure the version is valid and available in the configured registry.`);
 		}
+
 		return resolvedVersion;
 	}
 
+	static async _getVersionSpec(version, {ui5HomeDir, cwd}) {
+		if (this._isSnapshotVersionOrRange(version)) {
+			const versionMatch = version.match(VERSION_RANGE_REGEXP);
+			if (versionMatch) {
+				// For snapshot version ranges we need to insert a stand-in "x" for the patch level
+				// and - in case none is provided - another "x" for the major version in order to
+				// convert it to a valid semver range:
+				// "1-SNAPSHOT" becomes "1.x.x-SNAPSHOT" and "1.112-SNAPSHOT" becomes "1.112.x-SNAPSHOT"
+				return `${versionMatch[1]}.${versionMatch[2] || "x"}.x-SNAPSHOT`;
+			}
+		}
+
+		if (semver.validRange(version)) {
+			return version;
+		}
+
+		// Check for invalid tag name (same check as npm does)
+		if (encodeURIComponent(version) !== version) {
+			return null;
+		}
+
+		const allTags = await this.fetchAllTags({ui5HomeDir, cwd});
+
+		if (!allTags) {
+			// Resolver doesn't support tags (e.g. Sapui5MavenSnapshotResolver)
+			// Only latest and latest-snapshot are supported which both resolve
+			// to the latest available version.
+			// See "isSnapshotVersionOrRange" for -snapshot handling
+			if ((version === "latest" || version === "latest-snapshot")) {
+				return "*";
```

Comment:
The asterisk (*) would resolve `latest`, but would this be correct for `latest-snapshot`?
What about `x.x.x-SNAPSHOT` for `latest-snapshot`?

Reply:
This is the existing behaviour. Using an asterisk and "includePrerelease" will resolve to the highest version.

`x.x.x-SNAPSHOT` sounds like a better fit, but it actually is handled the same by semver (`require("semver").validRange("x.x.x-SNAPSHOT") === "*"`).
Therefore, if there would be a higher released version (no -SNAPSHOT), it would always be picked, even with that range.

Usually that is not the case as the highest version always has a snapshot version available. But even if this wouldn't be the case, our snapshot resolver filters for -SNAPSHOT versions so it will always resolve to the highest snapshot.